Fixed issue #20828 Newsletter Problem Grid Filter issue

GovindaSharma opened the pull request:
### Description (*)
Fixed issue #20828 Newsletter Problem Grid Filter issue

### Fixed Issues (if relevant)

1. magento/magento2 #20828 Newsletter Problem Grid Filter issue


### Manual testing scenarios (*)

1.Go To Magento Backend 
2.Navigate to Reports->Newsletter Problem Reports
3.Grid Will Open
4.Click on Search or Reset Filter Button

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
Newsletter Grid Filter fix

magento-engcom-team commented:
Hi @GovindaSharma. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

VladimirZaets commented on app/code/Magento/Newsletter/Controller/Adminhtml/Problem/Index.php:
```diff
@@ -6,9 +6,10 @@
 use Magento\Framework\App\Action\HttpGetActionInterface as HttpGetActionInterface;
 
-class Index extends \Magento\Newsletter\Controller\Adminhtml\Problem implements HttpGetActionInterface
+class Index extends \Magento\Newsletter\Controller\Adminhtml\Problem implements HttpGetActionInterface, HttpPostActionInterface
```
Hi @GovindaSharma. Thanks for collaboration. Can you please explain how your changes related to the bug in the issue?

GovindaSharma replied:
Hello @VladimirZaets ,as when grid is loading then it is get request but when you click on Search or Rest Filter button it becomes post request,but in controller only HttpGetActionInterface is implemented because of this filters not working,so i added the HttpPostActionInterface ,so whenever filters is called it is handled by HttpPostActionInterface.

You can check in Newsletter queue Grid,there it is working fine because in the controller(app/code/Magento/Newsletter/Controller/Adminhtml/Queue/Index.php) both HttpPostActionInterface and HttpGetActionInterface is implemented.

VladimirZaets replied:
@GovindaSharma Hi, I understand the case. So, the code that currently exists in Magento will be refactored (or you can do it also in another PR if you want).

The right way to resolve this issue is creating additional POST controller and use it for filters operations.

GovindaSharma replied:
@VladimirZaets ok then i will change accordingly,since controller is already created we just need to pass that controller url.I will update you on this.
Thank You

Commit:
Reverted changes

Commit:
added grid url

GovindaSharma commented:
@magento-engcom-team give me test instance

magento-engcom-team commented:
Hi @GovindaSharma. Thank you for your request. I'm working on Magento instance for you

magento-engcom-team commented:
Hi @GovindaSharma, here is your [new Magento instance](https://pr-20829.instances.magento-community.engineering).
Admin access: [https://pr-20829.instances.magento-community.engineering/admin](https://pr-20829.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

GovindaSharma commented:
@VladimirZaets i have updated the changes according to your request.Reverted the previous changes and just added the grid url  in xml,now all filters request will directly go to grid url. Please check and update on this.

magento-engcom-team commented:
Hi @VladimirZaets, thank you for the review.
[ENGCOM-4179](https://jira.corp.magento.com/browse/ENGCOM-4179) has been created to process this Pull Request
